File: PythonProject/toString/CloneExample.py
import copy

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def clone(self):
        return copy.copy(self)

def main():
    # Create an original object
    original = Person("Alice", 25)

    # Create a clone of the original object
    cloned = original.clone()

    # Display original and cloned object details
    print(f"Original: {original.name}, {original.age}")
    print(f"Cloned: {cloned.name}, {cloned.age}")

if __name__ == "__main__":
    main()

File: JavaProject/src/com/toString/CloneExample.java
package com.toString; //package

import com.loggerutilities.LoggerUtility; //import logger utility
import java.util.logging.Logger;
import java.util.logging.Level;

class Person11 implements Cloneable { // Implement Cloneable interface
    String name;
    int age;

    // Constructor
    Person11(String name, int age) {
        this.name = name;
        this.age = age;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return super.clone(); // Call the clone() method from Object class
    }
}

public class CloneExample {
    private static final Logger logger = LoggerUtility.getLogger(); //get logger from LoggerUtility

    public static void main(String[] args) {
        try {
            // Create an original object
            Person11 original = new Person11("Alice", 25);

            // Create a clone of the original object
            Person11 cloned = (Person11) original.clone();

            // Display original and cloned object details
            System.out.println("Original: " + original.name + ", " + original.age);
            System.out.println("Cloned: " + cloned.name + ", " + cloned.age);

        } catch (CloneNotSupportedException e) {
            logger.log(Level.SEVERE, "CloneNotSupportedException caught: " + e.getMessage(), e); //log exception
            System.out.println("Cloning failed: " + e.getMessage()); //print message to console
        } catch (Exception e) {
            logger.log(Level.SEVERE, "Unexpected error: " + e.getMessage(), e); //log unexpected error
            System.out.println("An unexpected error occurred: " + e.getMessage()); //print message to console
        }
    }
}
